cmake/faceliftMacros.cmake

```diff
@@ -136,67 +136,103 @@ function(facelift_generate_code )
 
-    if(NOT DEFINED ENV{LANG}) # e.g. Qt Creator was started from the Apple Dock
-        set(ENV{LANG} "en_US.UTF-8")
-        # --> http://click.pocoo.org/5/python3/#python-3-surrogate-handling
-        # --> https://apple.stackexchange.com/questions/54765/how-can-i-have-qt-creator-to-recognize-my-environment-variables
-        set(LANG_SET_BY_FACELIFT true)
-    else()
-        set(LANG_SET_BY_FACELIFT false)
-    endif()
+        if(NOT DEFINED ENV{LANG}) # e.g. Qt Creator was started from the Apple Dock
+            set(ENV{LANG} "en_US.UTF-8")
+            # --> http://click.pocoo.org/5/python3/#python-3-surrogate-handling
+            # --> https://apple.stackexchange.com/questions/54765/how-can-i-have-qt-creator-to-recognize-my-environment-variables
+            set(LANG_SET_BY_FACELIFT true)
+        else()
+            set(LANG_SET_BY_FACELIFT false)
+        endif()
 
-    execute_process(COMMAND ${FACELIFT_PYTHON_EXECUTABLE} ${BASE_CODEGEN_COMMAND}
-        RESULT_VARIABLE CODEGEN_RETURN_CODE
-        WORKING_DIRECTORY ${QFACE_BASE_LOCATION}/qface
-        OUTPUT_VARIABLE CODEGEN_OUTPUT
-        ERROR_VARIABLE CODEGEN_ERROR
-    )
+        execute_process(COMMAND ${FACELIFT_PYTHON_EXECUTABLE} ${BASE_CODEGEN_COMMAND}
+            RESULT_VARIABLE CODEGEN_RETURN_CODE
+            WORKING_DIRECTORY ${QFACE_BASE_LOCATION}/qface
+            OUTPUT_VARIABLE CODEGEN_OUTPUT
+            ERROR_VARIABLE CODEGEN_ERROR
+        )
 
-    if(LANG_SET_BY_FACELIFT)
-        unset(ENV{LANG})
-    endif()
+        if(LANG_SET_BY_FACELIFT)
+            unset(ENV{LANG})
+        endif()
 
-    if(NOT "${CODEGEN_RETURN_CODE}" STREQUAL "0")
-        message(FATAL_ERROR "Facelift code generation failed!\nCommand: ${BASE_CODEGEN_COMMAND_WITH_SPACES} with PYTHONPATH=$ENV{PYTHONPATH}\nReturn code: ${CODEGEN_RETURN_CODE}\nOutput: ${CODEGEN_OUTPUT}\nError: ${CODEGEN_ERROR}\n")
-    endif()
+        if(NOT "${CODEGEN_RETURN_CODE}" STREQUAL "0")
+            message(FATAL_ERROR "Facelift code generation failed!\nCommand: ${BASE_CODEGEN_COMMAND_WITH_SPACES} with PYTHONPATH=$ENV{PYTHONPATH}\nReturn code: ${CODEGEN_RETURN_CODE}\nOutput: ${CODEGEN_OUTPUT}\nError: ${CODEGEN_ERROR}\n")
+        endif()
 
-    facelift_synchronize_folders(${WORK_PATH} ${ARGUMENT_OUTPUT_PATH})
+        facelift_synchronize_folders(${WORK_PATH} ${ARGUMENT_OUTPUT_PATH})
 
-    # Delete work folder
-    file(REMOVE_RECURSE ${WORK_PATH})
+        # Delete work folder
+        file(REMOVE_RECURSE ${WORK_PATH})
+    else()
+        message("No interface files modified for target ${LIBRARY_NAME}. Code generation not started.")
+    endif()
 
     # Add a dependency so that CMake will reconfigure whenever one of the interface files is changed, which will refresh our generated files
-    file(GLOB_RECURSE QFACE_FILES ${ARGUMENT_INTERFACE_DEFINITION_FOLDER}/*.qface)
     file(GLOB_RECURSE CODEGEN_FILES ${CODEGEN_LOCATION}/*)
```

Comment:
A change to any of these files should also cause the code generator to be executed.